src/exe-elf-arm-fixup.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright © 2022 László "MrAnno" Várady
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef EXE_ELF_ARM_FIXUP_H_
+#define EXE_ELF_ARM_FIXUP_H_
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "common.h"
+#include "exe-elf.h"
+
+extern void *bxfi_trampoline_thumb;
+extern void *bxfi_trampoline_thumb_addr;
+extern void *bxfi_trampoline_thumb_end;
+
+extern void *bxfi_trampoline_thumb_nop;
+extern void *bxfi_trampoline_thumb_nop_end;
+
+static inline int bxfi_exe_is_arm_thumb_func(void *func_addr)
+{
+    return (uintptr_t) func_addr & 0x1U;
+}
+
+/* The returned address should not be called, it is meant to be used for patching */
+static inline void bxfi_exe_fix_func_addr_if_in_arm_thumb_mode(void **addr)
+{
+    *addr = (void *) (uintptr_t) ((uintptr_t) *addr & ~0x1ULL);
+}
+
+static inline int bxfi_exe_is_arm_func_word_aligned(void *func_addr)
+{
+    return ((uintptr_t) func_addr & 0x3U) == 0;
+}
+
+static inline void bxfi_exe_trampoline_fixup(void **func_to_patch, void **trampoline,
```

Comment:
I think I'd rather have both this and bxfi_exe_inject_prelude be static functions in exe-elf.c (or some other .c file) and then use the preprocessor to do things conditionally based on the architecture.

Reply:
Actually, isn't this prelude patching also required for Windows ARM? Surely the thumb issues are also a thing over there.

Reply:
Oh, I did not know about Windows on ARM.

I've renamed the unit to `exe-trampoline-fixup`. Hopefully, the code is general enough to be used for the Windows ARM support too. I'd add Windows support in a separate PR, if that is OK with you.

Reply:
Yes, I'm fine with Windows support being added in another PR. It's also worth noting that it's likely that the same workaround will need to be ported to exe-mach-o.c too if anyone wants to make use of BoxFort on some iOS platforms, though I don't think there's much of an audience for that use-case.